### PR TITLE
Update the main documentation.

### PR DESCRIPTION
**Pull request by cgruber**
A series of changes adding to the public documentation, and reorganizing it slightly.

### CODE REVIEW

File: producers.md [hunk 1]
```diff
@@ -242,9 +244,40 @@ subcomponents of either components or production components.
...
-### Logging and debugging
+### Monitoring
 
-==As of April 2015, not implemented yet==
+[ProducerMonitor][ProducerMonitor] can be used to monitor the execution of
+producer methods; its methods correspond to various places in a producer's
+lifecycle.
+
+To install a `ProducerMonitor`, contribute into a set binding of
+[ProductionComponentMonitor.Factory][ProductionComponentMonitorFactory]. For
```
Comment (tbroyer) [inline producers.md]: backticks around `ProductionComponentMonitor.Factory`?

File: sitemap.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
  * [Home](index.md)
  * [User's Guide](users-guide.md)
+   * [Dagger & Android](android.md)
```
Comment (ronshapiro) [inline sitemap.md]: Doesn't look like this got synced out

File: sitemap.md [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
    * [Multibindings](multibindings.md)
    * [Subcomponents](subcomponents.md)
    * [Migrating from Dagger 1](dagger-1-migration.md)
    * [Producers](producers.md)
+   * [Testing](testing.md)
```
Comment (ronshapiro) [inline sitemap.md]: This too

File: sitemap.md [hunk 4]
```diff
@@ -1,8 +1,10 @@
  * [Home](index.md)
  * [User's Guide](users-guide.md)
```
Comment (ronshapiro) [inline sitemap.md]: Also the users-guide.md

File: sitemap.md (hunk 2 above)
Comment (cgruber) [inline sitemap.md]: That's true, but it was never added to the sync tarball.  That can happen later.  It's irrelevant to this bit, though, insofar as this sitemap.md isn't used in the gh-pages jekyll template.

File: producers.md (hunk 1 above)
Comment (cgruber) [inline producers.md]: Have to resolve separately, as this is a sync CL.  It's a good note, though.  Feel free to throw it into a Pull Request, if you wish, as we can yank pull requests in more easily now.  But these sync pulls I don't want to alter directly, in that they reflect a 1:1 with already recorded commits internally.

File: sitemap.md (hunk 3 above)
Comment (cgruber) [inline sitemap.md]: Ditto.  We can follow up.  I'm trying to figure out a way of getting a jekyll template that can interpret these sitemap/navbar md files, but for now, these are strictly irrelevant to the open-source docs.

File: sitemap.md (hunk 4 above)
Comment (cgruber) [inline sitemap.md]: Yep.  This has to be fixed forward, and ditto.

Comment (cgruber) [inline sitemap.md]: Actually, this has already been synced out and just didn't have diffs.

Comment (ronshapiro) [inline sitemap.md]: Whoops, that's already been out.

File: multibindings.md [hunk 5]
```diff
@@ -329,6 +315,48 @@ class MyProviderMapModule {
...
+## Declaring multibindings
+
+You can declare that a multibound set or map is bound by nesting a
+[`@Multibindings`]-annotated interface within a module, with methods that return
+the sets or maps you want to declare.
+
+You do not have to use [`@Multibindings`] for sets or maps that have at least
+one contribution, but you do have to declare them if they may be empty.[^empty-set-values]
+
+[^empty-set-values]:
```
Comment (ronshapiro) [inline multibindings.md]: @netdpb this syntax doesn't appear to work in github-flavored markdown

File: producers.md (hunk 1 above)
Comment (ronshapiro) [inline producers.md]: I just sent this to @cgruber this internally.

**Pull request merged**